firmware-management/mbl-watchdog-init/mbl-watchdog-init.c

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2019 Arm Limited and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ *
+ * This utility initialises the hardware watchdog and sets the timeout.
+ * The program also tries to determine the last boot reason so we can tell if
+ * the last reboot was caused by the hardware watchdog.
+ * This code is intended for use with our robust update mechanism; we need
+ * to start the watchdog before we mount the rootfs, so the device reboots if
+ * loading the rootfs fails.
+ */
+
+#include <errno.h>
+#include <fcntl.h>
+#include <getopt.h>
+#include <linux/watchdog.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/ioctl.h>
+#include <unistd.h>
+
+//============================================================================
+// Error handling and logging convenience functions
+int is_err(const int ret)
+{
+    return ret == -1;
+}
+
+
+int log_error(const char *message)
+{
+    int err_code = errno;
+    char *prefix = "WATCHDOG-ERROR:";
+    char result[strlen(prefix) + strlen(message)];
+
+    snprintf(result, sizeof(result), "%s %s", prefix, message);
+    perror(result);
+
+    return err_code;
+}
+
+
+void log_warning(const char *message)
+{
+    char *prefix = "WATCHDOG-WARNING:";
+    printf("%s %s\n", prefix, message);
+}
+
+
+void log_info(const char *message)
+{
+    char *prefix = "WATCHDOG-INFO:";
+    printf("%s %s\n", prefix, message);
+}
+
+//============================================================================
+// Watchdog functions
+int set_watchdog_timeout(const int watchdog_fd, int *timeout)
+{
+    return ioctl(watchdog_fd, WDIOC_SETTIMEOUT, timeout);
+}
+
+
+int get_last_boot_status(const int watchdog_fd, int *flags)
+{
+    return ioctl(watchdog_fd, WDIOC_GETBOOTSTATUS, flags);
+}
+
+
+int print_last_boot_reason(const int boot_status)
+{
+    switch (boot_status)
+    {
+        case WDIOF_OVERHEAT:
+            log_info("The last reboot was caused by the CPU overheating.\n");
+            return 0;
+        case WDIOF_CARDRESET:
+            log_info("The last reboot was caused by a watchdog reset.\n");
+            return 0;
+    }
+    return -1;
+}
+
+//============================================================================
+// Main entry point
+int main(int argc, char **argv)
+{
+    int timeout = 0;
+    char* wdog_filename = "";
+
+    // Parse command line arguments
+    int current_opt;
+    int optindex;
+    struct option longopts[] = {
+        {"timeout", required_argument, NULL, 't'},
+        {"wdog-fname", required_argument, NULL, 'w'},
```

Comment:
```suggestion
        {"device", required_argument, NULL, 'd'},
```
I think that using the `device` wording is more intuitive.

Reply:
728ec9e